### PR TITLE
Create attorneys, claims_agents, representatives, organizations, and organizations_representatives tables

### PR DESCRIPTION
**Pull request by holdenhinkle**
Resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/78731

## Summary
This PR creates the following new tables:
- accredited_organizations
- accredited_representatives
- accredited_organizations_accredited_representatives
- accredited_attorneys
- accredited_claims_agents

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/78731

## Testing done
- NA

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
The database
The schema.rb file

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: db/migrate/20240405184242_create_orgs_reps_tables.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      t.float :lat
+      t.float :long
+      t.geography :location, limit: { srid: 4326, type: 'st_point', geographic: true }
+      t.timestamps
+    end
+    add_index :accredited_organizations, :location, using: :gist
+    add_index :accredited_organizations, :name
+
+    create_join_table :accredited_organizations, :accredited_representatives do |t|
+      t.index :accredited_organization_poa_code
```
Comment (jvcAdHoc) [inline db/migrate/20240405184242_create_orgs_reps_tables.rb:132]: I wonder if `t.references :accredited_organization, index: true, foreign_key: true` would do the same as this line + manually adding the foreign key?

File: db/migrate/20240405184242_create_orgs_reps_tables.rb [hunk 2]
```diff
@@ -0,0 +1,146 @@
+# frozen_string_literal: true
+
+class CreateOrgsRepsTables < ActiveRecord::Migration[7.1]
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/AbcSize
+  def change
+    create_table :accredited_attorneys, id: false, primary_key: :registration_number do |t|
+      t.string :registration_number, null: false
```
Comment (jvcAdHoc) [inline db/migrate/20240405184242_create_orgs_reps_tables.rb:8]: Thoughts on using the OGC uuid as the id?

File: db/migrate/20240405184242_create_orgs_reps_tables.rb (hunk 1 above)
Comment (holdenhinkle) [reply db/migrate/20240405184242_create_orgs_reps_tables.rb]: I update the migration to try that.

File: db/migrate/20240405184242_create_orgs_reps_tables.rb (hunk 2 above)
Comment (holdenhinkle) [reply db/migrate/20240405184242_create_orgs_reps_tables.rb:8]: Right!

**Comment by holdenhinkle** [comment]
@jvcAdHoc It appears the join table was setup correctly:

```
vets-api=# \d accredited_organizations_accredited_representatives
  Table "public.accredited_organizations_accredited_representatives"
            Column            | Type | Collation | Nullable | Default 
------------------------------+------+-----------+----------+---------
 accredited_organization_id   | uuid |           |          | 
 accredited_representative_id | uuid |           |          | 
Indexes:
    "idx_on_accredited_organization_id_4df677e34d" btree (accredited_organization_id)
    "idx_on_accredited_representative_id_c58ba2a35b" btree (accredited_representative_id)
    "index_organization_representatives_on_rep_and_org" UNIQUE, btree (accredited_organization_id, accredited_representative_id)
Foreign-key constraints:
    "fk_rails_02a6d27eb4" FOREIGN KEY (accredited_representative_id) REFERENCES accredited_representatives(id)
    "fk_rails_e84adfaa8f" FOREIGN KEY (accredited_organization_id) REFERENCES accredited_organizations(id)
```

File: db/migrate/20240405184242_create_orgs_reps_tables.rb [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    end
+    add_index :accredited_representatives, :registration_number, unique: true
+    add_index :accredited_representatives, :full_name
+    add_index :accredited_representatives, :location, using: :gist
+
+    create_table :accredited_organizations, id: :uuid do |t|
+      t.string :poa_code, limit: 3, null: false
+      t.string :name
+      t.string :phone
+      t.string :state, limit: 2
```
Comment (jvcAdHoc) [inline db/migrate/20240405184242_create_orgs_reps_tables.rb:112]: This is likely copy+paste from the original table. We only need `state_code`.

File: db/migrate/20240405184242_create_orgs_reps_tables.rb [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+      t.geography :location, limit: { srid: 4326, type: 'st_point', geographic: true }
+      t.timestamps
+    end
+    add_index :accredited_organizations, :location, using: :gist
+    add_index :accredited_organizations, :name
+
+    # Use create_table (instead of create_join_table) to explicitly define the table and its columns
+    create_table :accredited_organizations_accredited_representatives, id: false do |t|
+      t.uuid :accredited_organization_id
+      t.uuid :accredited_representative_id
```
Comment (jvcAdHoc) [inline db/migrate/20240405184242_create_orgs_reps_tables.rb:139]: I think we want `null: false` on these

**Review by jvcAdHoc — APPROVED** [review]

**Comment by stevenjcumming** [comment]
@holdenhinkle Once you resolve the merge conflict, I'll approve the PR

**Comment by holdenhinkle** [comment]
all set @stevenjcumming Thanks!

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**